Set PAM_TTY for remote displays too (used e.g. by pam_time)
--- xdm-1.1.1/greeter/greet.c.orig	2006-10-19 01:55:50.000000000 +0000
+++ xdm-1.1.1/greeter/greet.c	2006-11-15 08:39:41.739505000 +0000
@@ -545,8 +545,7 @@
 					(*pamhp, PAM_RHOST, hostname));
 		free(hostname);
 	    }
-	} else
-	    RUN_AND_CHECK_PAM_ERROR(pam_set_item, (*pamhp, PAM_TTY, d->name));
+	}
 
 	if (!greet->allow_null_passwd) {
 	    pam_flags |= PAM_DISALLOW_NULL_AUTHTOK;
